modules/core/src/test/scala/catcheffect/IOCatchTest.scala

```diff
@@ -0,0 +1,68 @@
+package catcheffect
+
+import catcheffect.Catch.RaisedWithoutHandler
+import catcheffect.IOCatch.ioCatching
+import cats.effect.*
+import munit.CatsEffectSuite
+
+class IOCatchTest extends CatsEffectSuite {
+  test("should abort execution as soon as raise is called") {
+    for {
+      ref <- Ref[IO].of("init")
+      res <- ioCatching[String](h =>
+        for {
+          _ <- h.raise("error")
+          _ <- ref.set("should not be set")
+        } yield 1
+      )
+      _ = assertEquals(res, Left("error"))
+      _ <- assertIO(ref.get, "init")
+    } yield ()
+  }
+  
+  test("can nest correctly") {
+    for {
+      res <- ioCatching[String](hs1 =>
+        for {
+          _ <- ioCatching[String](hs2 =>
+            hs1.raise("1")
+          )
+          _ <- IO.raiseError(new AssertionError("should not reach this point"))
+        } yield ()
+      )
+      _ = assertEquals(res, Left("1"))
+    } yield res
+  }
+  
+  test("Can raise in an uncancellable region") {
+    for {
+      res <- ioCatching[String](h =>
+        IO.uncancelable(_ => h.raise("oops"))
+      )
+      _ = assertEquals(res, Left("oops"))
+    } yield ()
+  }
+
+  test("Can raise in an cancellable region when polled") {
+    for {
+      res <- ioCatching[String](h =>
+        IO.uncancelable(poll => poll(h.raise("oops")))
+      )
```

Comment:
This one is interesting because I couldn't get the RaisedInUncancellable exception to be thrown unless I remove the `race` in https://github.com/ValdemarGr/catch-effect/blob/7613376a622864269cdd18c904039c5cad8fd6fc/modules/core/src/main/scala/catcheffect/Catch.scala#L267

Perhaps there's a race condition and somehow the outer cancellation always wins? :thinking: 
Either way, we will need to use raiseError to abort the execution anyway so nothing changes.

Reply:
Yeah I just tried `prom.get <* F.sleep(1.second)` and it raised the error.

I think it is very hard to have it throw consistently without being able to query cats-effect about the "current cancelation" context.

Unfortunately I think that also means that the test "Can raise in an uncancellable region" might be flaky :crying_cat_face:

Reply:
I guess we can make both cases pass for now since it's race-y :man_shrugging: Will need to document this behaviour